seatunnel-core/seatunnel-spark-starter/seatunnel-spark-starter-common/src/main/java/org/apache/seatunnel/core/starter/spark/execution/TransformExecuteProcessor.java

```diff
@@ -101,7 +102,7 @@ public List<DatasetTableInfo> execute(List<DatasetTableInfo> upstreamDataStreams
                 TableTransformFactory factory = plugins.get(i);
                 TableTransformFactoryContext context =
```

Comment:
Maybe you can use `dataset.getCatalogTables()` instead of `Collections.singletonList(dataset.getCatalogTables().get(0))` ?. Like this:
```java
TableTransformFactoryContext context =                        
    new TableTransformFactoryContext(
        dataset.getCatalogTables(),
        ReadonlyConfig.fromConfig(pluginConfig),
        classLoader);
```